source/PlayerEntity.cpp

```diff
@@ -90,21 +90,46 @@ void PlayerEntity::onPose(Array<shared_ptr<Surface> >& surfaceArray) {
 }
 
 void PlayerEntity::updateFromInput(UserInput* ui) {
+	
+	float walkSpeed;
 
-	const float walkSpeed = *moveRate * units::meters() / units::seconds();
+	// Check if player is sprinting or not
+	if (!m_sprinting) {
+		walkSpeed = *moveRate * units::meters() / units::seconds();
+	}
+	else {
+		walkSpeed = *moveRate * *sprintMultiplier * units::meters() / units::seconds();
+	}
 
 	// Get walking speed here (and normalize if necessary)
-	Vector3 linear = Vector3(ui->getX()*moveScale->x, 0, -ui->getY()*moveScale->y);
-	if (linear.magnitude() > 0) {
-		linear = linear.direction() * walkSpeed;
+	m_linearVector = Vector3(ui->getX()*moveScale->x, 0, -ui->getY()*moveScale->y);
+	if (m_linearVector.magnitude() > 0) {
+		m_acceleratedVelocity = lerp(m_acceleratedVelocity, walkSpeed, *movementAcceleration);
```

Comment:
What does this imply the units of acceleration are? We should certainly document acceleration w/ correct units. It might be easier to apply a m/s^2 acceleration here as opposed to doing a lerped value. My concern here is that this makes acceleration/deceleration tied to frame rate (number of times this function is called per second).

Reply:
Now the acceleration is not tied to the frame rate. Lerp will be controlled by Delta-Time.